kratos/utilities/search_utilities.cpp

```diff
@@ -21,6 +21,56 @@
 namespace Kratos
 {
 
+bool SearchUtilities::PointIsInsideBoundingBox(
+    const BoundingBox<Point>& rBoundingBox,
+    const array_1d<double, 3>& rCoords
+    )
+{
+    // Get the bounding box points
+    const auto& r_max_point = rBoundingBox.GetMaxPoint();
+    const auto& r_min_point = rBoundingBox.GetMinPoint();
+
+    // The Bounding Box check
+    if (rCoords[0] < r_max_point[0] && rCoords[0] > r_min_point[0])           // check x-direction
+        if (rCoords[1] < r_max_point[1] && rCoords[1] > r_min_point[1])       // check y-direction
+            if (rCoords[2] < r_max_point[2] && rCoords[2] > r_min_point[2])   // check z-direction
+                return true;
```

Comment:
maybe there would be a way to unify this logic, since it seems prone to copy-paste errors :thinking:

Reply:
Yep, we could